### DISCUSSION TITLE
How could I get the intermediate result in the first stage and use it in the second stage?

### DISCUSSION THREAD

**Discussion by hubutui** [Project-MONAI/MONAILabel#1503] [Q&A] [answered]
I would like to create a pipeline that the first stage uses hovernet from https://github.com/Project-MONAI/MONAILabel/tree/0.7.0/sample-apps/pathology to get segmentation result, and then in the second stage, I need the `instance_map` from the hovernet in the first stage. By default, the InferTask return `json, json_file`, and I found that I could add `skip_writer=True` when I create the `InferTask`, and the data dict is returned which contains the `instance_map` I need. However, according to https://github.com/Project-MONAI/MONAILabel/blob/0.7.0/monailabel/interfaces/app.py#L307, `skip_writer` is not respected here. Any idea without modifying monailbale itself?

**Comment by diazandr3s** [comment]
Hi @hubutui,

This is a great question.

Once you create the infer and train files for each of the stages, you should create an infer file for this pipeline. Here is an example of a pipeline we did for vertebra segmentation: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/infers/vertebra_pipeline.py#L135

That pipeline has 3 stages.

Here is another example of a 2-stages approach: https://github.com/Project-MONAI/MONAILabel/blob/KidneyMultiStage/sample-apps/radiology/lib/infers/kidney_pipeline.py

Hope this serves as an inspiration for a pipeline in pathology app

**Comment by hubutui** [comment, marked as answer]
Thanks, @diazandr3s. I got an example and found I just need to rewrite `writer()` method to return the data I need when the InferTask is in `pipeline_mode`. https://github.com/dgmato/MONAILabel/blob/fee2f92d3e9c51a8a7acf11a288d66cf6805801b/sample-apps/radiology/lib/infers/segmentation_teeth.py#L112-L114

**Comment by diazandr3s** [comment]
Superb! Glad to hear this.
